src/android/toga_android/widgets/label.py

```diff
@@ -1,40 +1,17 @@
-from android.view import Gravity
+from .base import Widget
+from ..libs.android_widgets import TextView
 
-from travertino.size import at_least
 
-from toga.constants import (
-    LEFT_ALIGNED,
-    RIGHT_ALIGNED,
-    CENTER_ALIGNED,
-    JUSTIFIED_ALIGNED,
-    NATURAL_ALIGNED,
-)
-
-
-class TogaLabel:
-    # TODO: Extend `android.widget.TextView`. Provide app as `context`.
-    def __init__(self, context, interface):
-        self.interface = interface
-
-
-class Label:
+class Label(Widget):
     def create(self):
-        self.native = TogaLabel(self.app.native, self.interface)
+        self.native = TextView(self._native_activity)
         self.native.setSingleLine()
 
-    def set_alignment(self, value):
-        self.native.setGravity({
-                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
-                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
-                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-            }[value])
-
     def set_text(self, value):
-        self.native.setText(self.interface._text)
+        self.native.setText(value)
 
     def rehint(self):
-        # print("REHINT label", self, self.native.getMeasuredWidth(), self.native.getMeasuredHeight())
-        self.interface.intrinsic.width = at_least(self.native.getMeasuredWidth() / self.app.device_scale)
```

Comment:
Interesting that you're not seeing any device scale issues - is that because you've got a device scale of 1? Mostly want to check we're not losing detail here because of an accident of test device selection.

Reply:
I don't really know, and I see the point you raise. :) I'm using the Android emulator, and it could well be device_scale is 1.

I would prefer to test (then fix) device scale in a follow-up PR. There may be real problems when testing that, but there is so much to do that I want to merge individual forward-progress PRs as much as possible. Testing is already somewhat slow for me (about a minute to start the app in the emulator), and I am reluctant to make it slower, but I can validate this if you like.

Note that the code I'm deleting is non-working code, so it's not really the case that we're _losing_ anything. `self.app.device_scale` would raise `AttributeError` at the moment, since my `App` doesn't have a `.device_scale` property at the moment.

Reply:
I'm mostly just interested in the extent to which this is "not seeing the problem because our test case doesn't reveal it". Windows DPI problems are the analog here; my test box isn't HiDPI, and every time I touch something on the Windows backend, I end up getting bug reports about rendering on HiDPI (because everyone's *actual* laptops have HiDPI turned on). 

Regardless, happy to handle this in a follow up.

Reply:
+1.

Maybe tomorrow's the day I set up a hidpi emulator (is that a thing? we'll see) too.